happi/containers.py

```diff
@@ -1,330 +1,30 @@
-    system by default.
-
-    We're simplifying here, assuming the LODCM is aligned and that the states
-    are accurate. We'll probably only look at the H1N state for lightpath
-    purposes, but it's good to collect all the available information.
-
-    Attributes
-    ----------
-    prefix : str
-        The base PV should be the state PV associated with the h1n or h2n
-        state, depending on which crystal we're referring to e.g.
-        "XPP:LODCM:H1N".
-
-    mono_line : str
-        Name of the mono line
-    """
-    device_class = copy(Device.device_class)
-    device_class.default = 'pcdsdevices.device_types.LODCM'
-    mono_line = EntryInfo("Name of the MONO beamline",
-                          enforce=str, optional=False)
-    kwargs = deepcopy(Device.kwargs)
-    kwargs.default.update({'mono_line': '{{mono_line}}',
-                           'main_line': '{{beamline}}'})
-
-
-class MovableStand(Device):
-    """
-    This class stores information about stands that move, like XPP's hand-crank
-    that moves SB2 and SB3 from the PINK to XPP lines and back. There is no
-    need to instantiate one of these for static stands.
-
-    Attributes
-    ----------
-    prefix : str
-        If there is a single PV with the stand's location, this should be the
-        base PV. In general, these devices will actually have multiple PVs
-        with binary outputs that have yes/no on the stand being in each
-        position. In these cases we pick the common prefix of these PVs.
-
-    stand : list
-        List of stands affected by table movement.
-    """
-    stand = copy(Device.stand)
-    stand.enforce = list
-    system = copy(Device.system)
-    system.default = 'changeover'
-
-
-class Motor(Device):
-    """
-    A Generic EpicsMotor
-    """
-    device_class = copy(Device.device_class)
-    device_class.default = 'pcdsdevices.device_types.Motor'
-    system = copy(Device.system)
-    system.default = 'motion'
-
-
-class AreaDetector(Device):
-    """
-    A Generic EpicsCamera
-    """
-    device_class = copy(Device.device_class)
-    device_class.default = 'pcdsdevices.device_types.PCDSDetector'
-    system = copy(Device.system)
-    system.default = 'camera'
-
-
-class Acromag(Device):
-    """
-    A Generic class for Acromag
-    """
-    device_class = copy(Device.device_class)
-    device_class.default = 'pcdsdevices.device_types.Acromag'
-    system = copy(Device.system)
-    system.default = 'acromag'
-
-
-class Trigger(Device):
-    """
-    A Generic class for Controls Triggers
-    """
-    device_class = copy(Device.device_class)
-    device_class.default = 'pcdsdevices.device_types.Trigger'
-    system = copy(Device.system)
-    system.default = 'timing'
+import logging
+import inspect
+import entrypoints
+
+from .item import HappiItem
+
+logger = logging.getLogger(__name__)
+
+_entries = entrypoints.get_group_all('happi.containers')
+
+registry = set()
```

Comment:
I wonder if the registry should be keyed as follows: `registry['(module).(classname)'] = obj`?

Reply:
What about the `name` from the entry_points? This way we can make it uniform to always be the package that originated it and we guarantee unique names per package?
E.g.:
`pcdsdevices.Acromag` as the container and key at registry.

Reply:
Hmm, maybe... but what about multi-level imports?

Would (e.g.) `pcdsdevices.motors.Acromag` become `pcdsdevices.Acromag`?

Reply:
The second one will be an internal key for Happi at the `registry` and it will point to the proper class `pcdsdevices.happi.containers.Acromag` and queried. I decided to make it shorter as a convenience for users when selecting a class at the CLI but we can also make the namespace shorter at the `pcdsdevices` side and improve the display of the possibilities at the CLI.

Reply:
In our example, the name you'd put in a happi entry to get the `Acromag` instantiated would be `pcdsdevices.Acromag`, right?

Reply:
That depends but I can see the confusion. Would it be okay if I make the namespace be `pcdsdevices.happi.Acromag` and I switch to use the full class at the registry like you suggested before?

Reply:
I'm trying to get at the heart of the problem and don't really know the "right" answer just yet to say yes or no...

`full_module.classname` is unambiguous. Entry_points `name` gives more flexibility, but adds indirection to get to the class itself.

Reply:
If we populate the registry with both `entry_point_name.classname` and `module.classname`, I can imagine that'd get confusing as well. I think it has to be one or the other.

The entrypoint name would allow for us to move around classes and refactor codebases without updating our device databases. Maybe that's preferable?

Reply:
As of now we have: 
```python
In [1]: import happi
Questionnaire backend unavailable: No module named 'psdm_qs_cli'

In [2]: happi.containers.registry
Out[2]:
{'pcdsdevices.Acromag': pcdsdevices.happi.containers.Acromag,
 'pcdsdevices.AreaDetector': pcdsdevices.happi.containers.AreaDetector,
 'pcdsdevices.Attenuator': pcdsdevices.happi.containers.Attenuator,
 'pcdsdevices.BeamControl': pcdsdevices.happi.containers.BeamControl,
 'pcdsdevices.Diagnostic': pcdsdevices.happi.containers.Diagnostic,
 'pcdsdevices.GateValve': pcdsdevices.happi.containers.GateValve,
 'pcdsdevices.IPM': pcdsdevices.happi.containers.IPM,
 'pcdsdevices.LCLSItem': pcdsdevices.happi.containers.LCLSItem,
 'pcdsdevices.LODCM': pcdsdevices.happi.containers.LODCM,
 'pcdsdevices.Motor': pcdsdevices.happi.containers.Motor,
 'pcdsdevices.MovableStand': pcdsdevices.happi.containers.MovableStand,
 'pcdsdevices.OffsetMirror': pcdsdevices.happi.containers.OffsetMirror,
 'pcdsdevices.PIM': pcdsdevices.happi.containers.PIM,
 'pcdsdevices.PulsePicker': pcdsdevices.happi.containers.PulsePicker,
 'pcdsdevices.Slits': pcdsdevices.happi.containers.Slits,
 'pcdsdevices.Stopper': pcdsdevices.happi.containers.Stopper,
 'pcdsdevices.Trigger': pcdsdevices.happi.containers.Trigger,
 'pcdsdevices.Vacuum': pcdsdevices.happi.containers.Vacuum}
```

So technically we are already doing that, right? Since you have the container class as the value of the dict you can get the `module.classname` ?
I agree, the entrypoint will allow for flexibility on this sense